rotkehlchen/db/dbhandler.py

```diff
@@ -3265,3 +3266,67 @@ def should_save_balances(self, cursor: 'DBCursor') -> bool:
         period = settings.balance_save_frequency * 60 * 60
         now = ts_now()
         return now - last_save > period
+
+    def get_web3_nodes(self, only_active: bool = False) -> List[WeightedNode]:
+        """
+        Get all the nodes in the database. If only_active is set to true only the nodes that
+        have the column active set to True will be returned.
+        """
+        with self.conn.read_ctx() as cursor:
+            if only_active:
+                cursor.execute('SELECT name, address, owned, weight, active FROM web3_nodes WHERE active=TRUE AND WEIGHT != "0.0";')  # noqa: E501
+            else:
+                cursor.execute('SELECT name, address, owned, weight, active FROM web3_nodes;')
+            return [
+                WeightedNode(
+                    node_info=NodeName(
+                        name=entry[0],
+                        endpoint=entry[1],
+                        owned=bool(entry[2]),
+                    ),
+                    weight=FVal(entry[3]),
+                    active=bool(entry[4]),
+                )
+                for entry in cursor
+            ]
+
+    def add_web3_node(self, node: WeightedNode) -> None:
+        with self.user_write() as cursor:
+            try:
+                cursor.execute(
+                    'INSERT INTO web3_nodes(name, address, owned, active, weight) VALUES (?, ?, ?, ?, ?)',   # noqa: E501
+                    node.serialize_for_db(),
+                )
+            except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
+                raise InputError(f'Node with name {node.node_info.name} already exists in db') from e  # noqa: E501
+
+    def update_web3_node(self, node: WeightedNode) -> None:
+        with self.conn.read_ctx() as cursor:
+            cursor.execute('SELECT COUNT(*) FROM web3_nodes WHERE name=?', (node.node_info.name,))
+            if cursor.fetchone() != (1,):
+                raise InputError(f'Node with name {node.node_info.name} doesn\'t exists')
+        with self.user_write() as cursor:
+            if node.weight == ZERO:
+                weight = '0.0'
+            else:
+                weight = str(node.weight)
```

Comment:
I made this to avoid having two different representations of the zero value

```
>>> str(FVal(0.0))
'0.0'
>>> str(FVal(0))
'0'
```

This happens because we do a `/100` from the information provided by the frontend and the `0.0` value can appear. I can change the check and do something as if the weight is ZERO then assign always str(ZERO)